tests/test_dataframe_show_reader.py

```diff
@@ -122,13 +191,32 @@ def test_show_output_to_df_when_values_are_null(
         assert 7 == len(rows[0])  # Number of columns
 
         row = rows[0]
-        assert None == row['string_column']
-        assert None == row['int_column']
-        assert None == row['bigint_column']
-        assert None == row['float_column']
-        assert None == row['timestamp_column']
-        assert None == row['default_type_column']
-        assert None == row['bool_column']
+        assert None is row['string_column']
+        assert None is row['int_column']
+        assert None is row['bigint_column']
+        assert None is row['float_column']
+        assert None is row['timestamp_column']
+        assert None is row['default_type_column']
+        assert None is row['bool_column']
+
+    def test_show_output_to_df_when_values_are_empty_and_data_types_are_specified(
+            self,
+            spark_session: SparkSession
+    ):
+        rows = show_output_to_df("""
+        +-------------+-------------------+
+        |string_column|default_type_column|
+        [string       |                   |
```

Comment:
This behavior has not changed. If the data type is `int` (whether specified via a data type declaration line or via the optional `default_data_type` param) and the value is not set, we get `ValueError: invalid literal for int() with base 10: ''`.

I have added tests to confirm and document this behavior for both the default data type and the data type declaration line cases. I think this is the safest behavior to prevent swallowing errors from unintentional oversights.

However, I could imagine in the future adding an optional `default_data_value` argument as a shorthand for someone who wants to, for example, initialize a grid to mostly `0` without having to put a `0` in each cell. I'm a little on the fence about adding that shorthand, as it might actually make it harder for someone inheriting the test to read it a glance.

Reply:
agreed, i'd rather force the user's hand than make a judgement based on likely intentions